lib/refinery_api.py

```diff
@@ -0,0 +1,131 @@
+            for {location}: {e}')
+    except (JSONDecodeError, KeyError) as e:
+        raise RefineryApiError(
+            f'Failed to parse Refinery API response: \
+                {type(e)} {e}')
+
+
+def parse_address(data):
+    return {
+        'line1': data.get('street_address'),
+        'city': data.get('locality'),
+        'state': data.get('region'),
+        'postal_code': data.get('postal_code')
+    }
+
+
+# given an array of fields and a location
+#  code, return an dict populated
+# by those fields for that location code.
+def get_refinery_data(code, fields):
+    data = {}
+    location = determine_location(code)
+    if location is None:
+        return None
+    logger.debug(f'Checking cache for {location}')
+    location_data = check_cache_and_or_fetch_data(location)\
+        .get('location_data')
+    if 'location' in fields:
+        data['location'] = parse_address(location_data)
+    if 'hours' in fields:
+        data['hours'] = build_hours_array(
+            location_data.get('hours').get('regular'),
+            datetime.datetime.today())
+    return data
+
+
+def check_cache_and_or_fetch_data(location):
+    location_data = fetch_location_data(location)
+    delta = datetime.datetime.today() - location_data.get('updated_at')
+    if delta.seconds > 3600:
+        fetch_location_data.cache_clear()
+        logger.info('Refreshing Refinery API data for location: ' + location)
+        location_data = fetch_location_data(location)
+    return location_data
+
+
+def determine_location(code):
+    location = None
+    if code.startswith('ma'):
+        location = 'schwarzman'
+    elif code.startswith('sc'):
+        location = 'schomburg'
+    elif code.startswith('pa'):
+        location = 'lpa'
+    else:
+        logger.warning('Unsupported location provided: ' + code)
+    return location
+
+
+# expects time - a string representing a 24hr clock time, and day - a
+# datetime object. Returns a
+#  string representing day updated to a new time
+# in format YYYY-MM-DDTHH:MM
+def build_timestamp(time, day):
+    if time is None:
+        return None
+    # extract 12 from '12:00'
+    hour = int(time.split(':')[0])
+    return day.replace(hour=hour).strftime('%Y-%m-%dT%X')
+
+
+# refinery day is one element of the hours array returned by Refinery API
+# today is a datetime object
+def build_hours_hash(refinery_day, offset, today):
+    new_date = today + datetime.timedelta(days=offset)
+    hours = {
+        'day': new_date.strftime('%A'),
+        'startTime': build_timestamp(
+            refinery_day.get('open'), new_date),
+        'endTime': build_timestamp(
+            refinery_day.get('close'), new_date),
+    }
+    if offset == 0:
+        hours['today'] = True
+    if offset == 1:
+        hours['nextBusinessDay'] = True
+
+    return hours
+
+
+def build_hours_array(days_array, today):
+    # put sunday
+    refinery_days_sunday_last = days_array[1:] + days_array[0:]
```

Comment:
This assumes Sunday will always be first in Refinery data. Can we rely on that?

Reply:
every response i've seen has had the same order. not sure if there is any written documentation that describes that behavior. i can add something to ensure it's in the correct order if you'd prefer that.

Reply:
Given that there is likely to be no further development on Refinery, let's assume it's a stable thing.